dispatcher/control.py

```diff
@@ -77,35 +82,45 @@ async def acontrol_with_reply(self, command: str, expected_replies: int = 1, tim
         except asyncio.TimeoutError:
             logger.warning(f'Did not receive {expected_replies} reply in {timeout} seconds, only {len(control_callbacks.received_replies)}')
             listen_task.cancel()
+        finally:
+            await broker.aclose()
 
         return self.parse_replies(control_callbacks.received_replies)
 
     async def acontrol(self, command: str, data: Optional[dict] = None) -> None:
         broker = get_broker(self.broker_name, self.broker_config, channels=[])
-        send_message = self.get_send_message(command=command, send_data=data)
-        await broker.apublish_message(message=send_message)
+        send_message = self.create_message(command=command, send_data=data)
+        try:
+            await broker.apublish_message(message=send_message)
+        finally:
+            await broker.aclose()
 
     def control_with_reply(self, command: str, expected_replies: int = 1, timeout: float = 1.0, data: Optional[dict] = None) -> list[dict]:
-        logger.info('control-and-reply {} to {}'.format(command, self.queuename))
+        logger.info(f'control-and-reply {command} to {self.queuename}')
         start = time.time()
         reply_queue = Control.generate_reply_queue_name()
-        send_message = self.get_send_message(command=command, reply_to=reply_queue, send_data=data)
+        send_message = self.create_message(command=command, reply_to=reply_queue, send_data=data)
 
         broker = get_broker(self.broker_name, self.broker_config, channels=[reply_queue])
 
         def connected_callback() -> None:
             broker.publish_message(channel=self.queuename, message=send_message)
 
         replies = []
-        for channel, payload in broker.process_notify(connected_callback=connected_callback, max_messages=expected_replies, timeout=timeout):
-            reply_data = json.loads(payload)
-            replies.append(reply_data)
-
-        logger.info(f'control-and-reply message returned in {time.time() - start} seconds')
-        return replies
+        try:
+            for channel, payload in broker.process_notify(connected_callback=connected_callback, max_messages=expected_replies, timeout=timeout):
+                reply_data = json.loads(payload)
+                replies.append(reply_data)
+            logger.info(f'control-and-reply message returned in {time.time() - start} seconds')
+            return replies
+        finally:
+            broker.close()
```

Comment:
I actually believe this is making the demo run slower now, and I'm sure would break things in production if it closes when in the middle of a transaction.

I'll work on pushing a commit to fix this. There's a flag that doesn't make any sense that I should get rid of. Now, we have a pretty clear sense of "ownership" of the connection. If the connection came from the factory we won't do anything to close it, because it's not owned by the broker.